src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,115 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] arrayList;
+
+    public ArrayList() {
+        arrayList = new Object[DEFAULT_CAPACITY];
+    }
+
+    private String errorOutOfBoundsMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+    private Object[] grow() {
+        int oldCapacity = arrayList.length;
+        if (oldCapacity > 0) {
+            int newCapacity = (oldCapacity >> 1) + oldCapacity;
+            return arrayList = Arrays.copyOf(arrayList, newCapacity);
+        } else {
+            return arrayList = new Object[DEFAULT_CAPACITY];
+        }
+    }
+
+    public void ensureCapacity() {
+        if (size + 1 >= arrayList.length) {
+            grow();
+        }
+    }
+
+    public Object[] toArray(List<T> list) {
+        Object[] listToArray = new Object[list.size()];
+        for (int i = 0; i < listToArray.length; i++) {
+            listToArray[i] = list.get(i);
+        }
+        return listToArray;
+    }
```

Comment:
let's better remove this method) 
or alternatively - if you still want to keep it - add it to interface as well